exec/java-exec/src/main/java/org/apache/calcite/jdbc/DynamicRootSchema.java

```diff
@@ -91,15 +91,17 @@ public void loadSchemaFactory(String schemaName, boolean caseSensitive) {
           return;
         }
 
-        // Found the storage plugin for first part(e.g. 'dfs') of schemaName (e.g. 'dfs.tmp')
-        // register schema for this storage plugin to 'this'.
-        plugin.registerSchemas(schemaConfig, thisPlus);
-
+        // Looking for the SchemaPlus for the top level (e.g. 'dfs') of schemaName (e.g. 'dfs.tmp')
+        SchemaPlus firstLevelSchema = thisPlus.getSubSchema(paths.get(0));
```

Comment:
What is thisPlus? Could we consider renaming it?

Reply:
It is also `SchemaPlus` instance which corresponds to the current schema. Thanks for pointing this, renamed `thisPlus` into `schemaPlus`.